File: neetcode/old/py/two_sum.py
# Given an array of integers num and an integer target, return the indices i and j such that
# nums[i] + nums[j] == target and i != j

from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dict = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_dict:
                return [num_dict[complement], i]
            num_dict[num] = i
        return None
    
solution = Solution()

nums = [3,4,5,6]
target = 7

print(solution.twoSum(nums, target))

File: neetcode/javascript/twoSum.ts
function twoSum(nums: number[], target: number): number[] {
  const numToIndex = new Map();
  const resultsArray = [];

  console.log('test');
  console.log(nums);

  for (let num of nums) {
    let complement = target - num;
    console.log(complement);

    if (numToIndex.has(complement)) {
      resultsArray.push(nums.indexOf(complement));
      resultsArray.push(nums.indexOf(num));
    }

    numToIndex.set(num, nums.indexOf(num));
  }

  console.log(numToIndex);
  console.log(resultsArray);

  return resultsArray;
}

let nums = [4, 5, 6];
let target = 10;

console.log(twoSum(nums, target));
